src/authenticationinapp/authenticationinapp.cpp

```diff
@@ -43,6 +46,13 @@ AuthenticationInApp::~AuthenticationInApp() {
 void AuthenticationInApp::setState(State state) {
   m_state = state;
   emit stateChanged();
+
+  const QMetaObject* meta = qt_getEnumMetaObject(state);
+  QMetaEnum me =
+      meta->enumerator(meta->indexOfEnumerator(qt_getEnumName(state)));
```

Comment:
this feels very heavy.

making sure i understand m_state is a number?

I have a recollection of doing work previously with things like 

`QString theBig = QVariant::fromValue(ModelApple::Big).toString();`

so that our debug messages didn't print out authentication state 0, 1, 2 but something human readable.

Can we do something nicer here.

Reply:
> I agree with "nicer". I disagree with "heavy".

I meant nicer :D This code has now gone away which is even better :D